geo/src/algorithm/rotate.rs

```diff
@@ -162,7 +162,10 @@ where
 {
     /// Rotate the LineString about its centroid by the given number of degrees
     fn rotate(&self, angle: T) -> Self {
-        rotate_many(angle, self.centroid().unwrap(), self.points_iter()).collect()
+        match self.centroid() {
+            Some(centroid) => rotate_many(angle, centroid, self.points_iter()).collect(),
+            None => self.clone(),
```

Comment:
This is only for the case of an empty linestring right?

Can you add a debug_assert or a comment to document this? Otherwise it might look like we're ignoring errors due to oversight.

Reply:
I've added a comment: happy to convert these to `debug_asserts` if we want to make it more prominent.